astropy/stats/sigma_clipping.py

```diff
@@ -558,7 +606,16 @@ def _sigmaclip_withaxis(
         else:
             return filtered_data
 
-    def __call__(self, data, axis=None, masked=True, return_bounds=False, copy=True):
+    def __call__(
+        self,
+        data: ArrayLike,
+        axis: int | tuple[int, ...] | None = None,
+        masked: bool | None = True,
+        return_bounds: bool | None = False,
+        copy: bool | None = True,
+    ) -> (
+        ArrayLike | tuple[ArrayLike, float, float] | tuple[ArrayLike, NDArray, NDArray]
```

Comment:
It's not actually ArrayLike, it's `NDArray | MaskedArray`.

Reply:
Fixed in [70b2007](https://github.com/astropy/astropy/pull/16562/commits/70b2007960c680ca2b474f8274f008b1f2030165)

Reply:
Fixed in [70b2007](https://github.com/astropy/astropy/pull/16562/commits/70b2007960c680ca2b474f8274f008b1f2030165)